test/performance/README.md

```diff
@@ -15,32 +15,81 @@ different rate, you can write your own pacer by implementing
-a Mako sidecar stub.
+## Testing architecture
 
-### Run without Mako
+The performance tests are based on Kubernetes Jobs running Golang code based on different [benchmarks](#benchmarks).
+The script [performance-tests.sh](./performance-tests.sh) first creates a cluster in GKE, installs Serving with specific settings
+for the performance tests. Then it installs the required Knative Services and runs the testing jobs.
 
-To run a benchmark once, and use the result from `mako-stub` for plotting:
+The results are written to:
+* Stdout
+* A logfile in a folder defined with env: $ARTIFACTS
+* To an InfluxDB hosted in the knative-community GKE project: https://github.com/knative/infra/tree/main/infra/k8s/shared
 
-1. Start the benchmarking job:
 
-   `ko apply -f test/performance/benchmarks/deployment-probe/continuous/benchmark-direct.yaml`
+## Grafana
 
-1. Wait until all the pods with the selector equal to the job name are completed.
+To better visualize the test results, Grafana is used to show some dashboards.
+The dashboards are defined in [grafana-dashboard.json](./influx/grafana-dashboard.json) and available under
 
-1. Retrieve results from mako-stub using the script in where `pod_name` is the name of the pod from the previous step.
+[perf.knative.dev](https://perf.knative.dev/d/igHJ5-gIk/knative-serving-performance-tests?orgId=1)
 
-  `read_results.sh "$pod_name" "$pod_namespace" ${mako_port:-10001} ${timeout:-120} ${retries:-100} ${retries_interval:-10} "$output_file"`
 
-   This will download a CSV with all raw results. Alternatively you can remove
-   the port argument `-p` in `mako-stub` container to dump the output to
-   container log directly.
+## Benchmarks
 
-**Note:** Running `performance-tests-mako.sh` creates a cluster and runs all the benchmarks in sequence. Results are downloaded in a temp folder
+Knative Serving has different benchmarking scenarios:
 
-### Benchmarking using Kperf
+* [dataplane-probe](./benchmarks/dataplane-probe): Measures the overhead Knative has compared to a `Deployment`
+* [load-test](./benchmarks/load-test): Measures request metrics for Knative Services under load in different scenarios (Activator always in path, Activator only in path at zero, Activator moving out of path on high-load) 
+* [reconciliation-delay](./benchmarks/reconciliation-delay): Measures the time it takes to reconcile a `KnativeService` and its child CRs.
+* [rollout-probe](./benchmarks/rollout-probe): Measures request metrics for a rolling update of a scaled `KnativeService`.
+* [scale-from-zero](./benchmarks/scale-from-zero): Measures the latency of scaling 1, 5, 25 and 100 Knative Services from zero in parallel.
 
-Running `performance-tests.sh` runs performance tests using [kperf](https://github.com/knative-extensions/kperf)
+
+## Running the benchmarks
+
+### Local influx setup
+
+You first need a local running instance of influxDB:
+
+```bash
+helm repo add influxdata https://helm.influxdata.com/
```

Comment:
Onstead of adding helm as a requirement users could install via yaml files directly as described in: https://docs.influxdata.com/influxdb/v2.7/install/?t=Kubernetes.

Reply:
Yes, same discussion as https://github.com/knative/serving/pull/14289#discussion_r1312908766 but in reverse :-). I'm fine with both ways, as this is just for local development anyways.

Reply:
I tried the approach with yaml, downside is that you need to configure users on the first start on the UI. So additional manual steps are necessary, which are not with helm. So I'd rather keep helm here, but point out that there is also a YAML option. WDYT?

Reply:
Yeah that is fine.